spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateProperties.java

```diff
@@ -133,6 +134,10 @@ private String determineDdlAuto(Map<String, String> existing, Supplier<String> d
 		if (ddlAuto != null) {
 			return ddlAuto;
 		}
+		String ddlDatabaseAction = existing.get(AvailableSettings.HBM2DDL_DATABASE_ACTION);
+		if (ddlDatabaseAction != null) {
+			return ddlDatabaseAction;
```

Comment:
The problem here is that `ddlDatabaseAction` is not, strictly speaking, a `ddlAuto`. The values are close but not the same. Considering that, I don't think that change is correct.

Rather, I think we should review this method altogether and set a ddl-auto only if necessary, calling the supplier if need to be. This could prove tricky as we've had a few regressions in this area.

Reply:
@snicoll I found the mapping from JPA (`javax.persistence.schema-generation.database.action`) to HBM (`hibernate.hbm2ddl.auto`) [here](https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#configurations-hbmddl). 

There is a direct mapping from each possible value of the JPA property to its corresponding HBM property. 

| JPA | HBM | Description | 
| --- | --- | ----------- | 
| none | none | No action will be performed |
| create | create-only | Database creation will be generated |
| drop | drop | Database dropping will be generated |
| drop-and-create | create | Database dropping will be generated followed by database creation |

The code in [CREATE_ONLY](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/tool/schema/Action.java#L29) and [CREATE](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/tool/schema/Action.java#L42) confirms this as well.

Since this ticket is about setting the HBM default value from the JPA value, I think the above grid gives us what we want to use. What do you think?

Reply:
> Since this ticket is about setting the HBM default value from the JPA value, I think the above grid gives us what we want to use. What do you think?

Unfortunately, I don't think this statement is accurate. IMO, this issue is about not setting a default ddlAuto value if the JPA spec property is set. I think translating one to the other is also a completely different feature that I am not convinced we want to provide.